src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -68,18 +68,29 @@ default String getAsTextHidePrivate() {
         final StringBuilder builder = new StringBuilder();
         builder.append(getName());
         if (!getPhone().isPrivate()) {
-            builder.append(" Phone: ").append(getPhone());
+            builder.append(getPrintableString(getPhone()));
         }
         if (!getEmail().isPrivate()) {
-            builder.append(" Email: ").append(getEmail());
+            builder.append(getPrintableString(getEmail()));
         }
         if (!getAddress().isPrivate()) {
-            builder.append(" Address: ").append(getAddress());
+            builder.append(getPrintableString(getAddress()));
         }
         builder.append(" Tags: ");
         for (Tag tag : getTags()) {
             builder.append(tag);
         }
         return builder.toString();
     }
+
+    /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+    default String getPrintableString(Printable... printables) {
+        String toBePrinted = "";
+        for(Printable printable: printables) {
+            toBePrinted += " " + printable.getPrintableString();
```

Comment:
You can do this using StringJoiner which makes your code "prettier" 
Read about it [here](https://stackoverflow.com/questions/27522563/why-stringjoiner-when-we-already-have-stringbuilder)